### PR TITLE
String utils for storing in game.prefs

### PR DESCRIPTION
**Pull request by 4z0t**
Solution for notifications
image
image
and lobby kick.
TODO:
* ~~apply to current problems~~
* ~~better implementation~~?

### CODE REVIEW

**Comment by Garanas** [comment]
What would being able to store strings as unicode introduce, feature-wise?

**Comment by Garanas** [comment]
Please ping me when this pull request is ready for review.

**Comment by 4z0t** [comment]
@Garanas I think it's time for review 🙂

**Review by Garanas — COMMENTED** [review]
I'm not entirely sure how often this code is run. But it is useful to upvalue calls to `math.mod`, `string.len` and others into (respectively) `local MathMod = math.mod` and `local StringLen = string.len` at the top of the file.

**Comment by Garanas** [comment]
How did you test these changes?

**Comment by 4z0t** [comment]
Just run game and see if it works correctly 😄

File: lua/UTF.lua [hunk 1]
```diff
@@ -59,3 +82,95 @@ function AddUnicodeCharToEditText(edit, unicode)
...
+    local function unicode_to_sym(unicode)
+        local bytes = {}
+        local unilen = StringLen(unicode)
+        for i = 3, unilen, 2 do
+            TableInsert(bytes, STR_xtoi(StringSub(unicode, i, i + 1)))
+        end
+        return StringChar(unpack(bytes))
+    end
+    local result = ''
+    for s in string.gfind(str, "\\u%x+") do
```
Comment (Garanas) [inline lua/UTF.lua:129]: Missed one

**Comment by KionX** [comment]
Perhaps all this is no longer necessary since the engine patch has been added: https://github.com/FAForever/FA-Binary-Patches/commit/b9329b4c218113b0f0d582d81ec2cb15b04cfeba

**Comment by 4z0t** [comment]
> Perhaps all this is no longer necessary since the engine patch has been added: https://github.com/FAForever/FA-Binary-Patches/commit/b9329b4c218113b0f0d582d81ec2cb15b04cfeba

Hope it is

**Pull request merged**